server/src/main/java/io/druid/curator/announcement/Announcer.java

```diff
@@ -217,6 +236,8 @@ public void announce(String path, byte[] bytes, boolean removeParentIfCreated)
                 @Override
                 public void childEvent(CuratorFramework client, PathChildrenCacheEvent event) throws Exception
                 {
+                  // NOTE: ZooKeeper does not guarantee that we will get every event, and thus PathChildrenCache doesn't
+                  // as well. If one of the below events are missed, Announcer might not work properly.
```

Comment:
This comment seems related to another one in the same file (last couple of sentences).

```
// Lost connection, which means session is broken, take inventory of what has been seen.
// This is to protect from a race condition in which the ephemeral node could have been
// created but not actually seen by the PathChildrenCache, which means that it won't know
// that it disappeared and thus will not generate a CHILD_REMOVED event for us.  Under normal
// circumstances, this can only happen upon connection loss; but technically if you have
// an adversary in the system, they could also delete the ephemeral node before the cache sees
// it.  This does not protect from that case, so don't have adversaries.
```

Reply:
Right. It's related.